Docs/update faq (#2287)
cryptoadvance/specter-desktop: docs/faq.md

## GENERAL QUESTIONS

### *How safe is the app to use?*

It is watch-only (private keys are protected by HWW) and compatible with multisig in Electrum, so even if something breaks you always have a fallback option while we fix the bug. So go for it :)

> We try to use default descriptors and derivation paths exactly for this reason - to be compatible with other wallets. Would be nice to keep it this way, but at a certain point we will need to diverge - for example when we add miniscript support.

Reviewer: ```suggestion
We try to use default descriptors and derivation paths exactly for this reason: to be compatible with other wallets. Even if we add advanced features such as miniscript, you will always have Bitcoin Core to fall back to. 
```

### *What's the difference between Specter-desktop and Specter-DIY?* 

Specter-desktop is a watch-only GUI software wallet running on Bitcoin Core or Electrum using its wallet and full node functionality.
Bitcoin Core tracks addresses, UTXO (unspent transaction outputs) and composes PSBT (partially-signed bitcoin transactions).

> Whereas, [Specter-DIY](https://docs.specter.solutions/diy/) is a do-it-yourself hardware wallet from off the shelf components, that signs and broadcasts transactions using QR codes that forgets your private keys when powered off.

Reviewer: ```suggestion
Whereas, [Specter-DIY](https://docs.specter.solutions/diy/) is a do-it-yourself hardware wallet from off the shelf-components, that signs transactions (mainly) using QR codes. You can configure it to forget your private keys when powering off or to store them (even on secure elements of smart cards). 
```

### *Is a full node necessary for using Specter-desktop?*

> No, a Bitcoin node is no longer needed. It's encouraged that you use your own Node (or your own electrum-server) but you can also use an external electrum-server in the public internet. Since Version 2 we have two proposals to use easily. As this was too cumbersome to use. we no longer provide an internal node since version 2. Use an external electrum instead.

Reviewer: ```suggestion
No, with Specter 2.0, a Bitcoin node is no longer needed. It's encouraged that you use your own Bitcoin Core node or your own Electrum server but you can also use a public Electrum server. Since version 2.0 we no longer support a Bitcoin Core node managed by the Specter app. You could use the external Electrum option instead or manage the Bitcoin Core node yourself.
```

### *I get this message: "This is a development server. Do not use it in a production deployment.". Should i worry?

No you don't have to worry unless you plan to run specter as a public service on the internet with a growing number of people using it. That was traditionally the use-case for web-apps: Run in the open public internet getting accessed by MANY people. However, that's not how specter is meant to be used. If people get performance issues, it's most likely not because the user-base is growing on their specter but because their Raspberry Pi is too weak for that one user who is doing all sorts of other stuff on that Pi on top.

### *I'm not sure I want the Bitcoin-Core wallet functionality to be used, is that mandatory? If so, is it considered secure?*

> If you use your own node and Bitcoin Core, You don't need private keys in Bitcoin Core, but you need wallets to be enabled `disablewallet=0` in your `bitcoin.conf` file. 

Reviewer: ```suggestion
If you use your own Bitcoin Core node, you don't need any private keys in Bitcoin Core, but you need wallets to be enabled (`disablewallet=0`) in your `bitcoin.conf` file. 
```

### How many addresses does an HD wallet have, and are they all the same?

By default the gap limit is 20, but you can go to the wallet settings and import as many addresses as you want. If you know the wallet is old you may want to try importing many addresses (~1000), and then rescanning.
The order is the same, and the addresses are also the same as the address derivation process is deterministic for a wallet. Address index is a derivation index of the wallet, so the index and the address itself are connected.

> Be aware that the number of addresses is impacting the sync-time if you use electrum.

Reviewer: That is true for Bitcoin Core as well.
```suggestion
Be aware that the number of addresses is impacting the sync-time.
```

The seed is generated by specter-desktop and then it's imported in a Bitcoin Core wallet, but instead of watch-only it's an XPRV imported using descriptors. More info on descriptors can be found [here](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md).

### *Why when I export a multisig wallet from specter-desktop (settings > export > copy wallet data) created from devices with only segwit ZPUBs, do I get a data structure with expected segwit derivation paths but XPUBs instead?*

XPUB is a canonical representation that is supported by Bitcoin Core, whereas ZPUB is an invention of SatoshiLabs that got adopted by the industry, but not by Bitcoin Core. In wallet export file we export Bitcoin Core's descriptor, so it contains master keys in the format that Bitcoin Core understands. More info on descriptors can be found [here](https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md).

### *Does Specter have coin control?*

> Yes, Specter supports coin control. Go to "Send". Open the "Advanced" features - and down at the right you have "Select coins manually" bottom. You can also choose UTXOs in the list and use them for creating a transaction.

Reviewer: ```suggestion
Yes, Specter supports coin control. Go to "Send". Below "Advanced", you can find "Coin selection", click on this. You can then choose UTXOs in the list manually and use them for creating a transaction.
```

Also you can run it via a platform-specific binary, pip-installation or via a Docker-container. It depends on your needs, and can be customized accordingly.

> Specter-desktop makes many requests to Bitcoin Core RPC (or Spectrum), so it works better from the same machine where Core is running, but remote is also possible. With that being said, by default Bitcoin Core RPC is connecting over HTTP, so everything including your RPC login and password are flying around as plaintext. You can use HTTPS and a [self-signed certificate](https://docs.specter.solutions/desktop/self-signed-certificates/) to fix that.

Reviewer: ```suggestion
Specter makes many requests to the Bitcoin Core RPC interface (or Spectrum), so it works better from the same machine where Core is running, but remote is also possible. With that being said, by default requests to the Bitcoin Core RPC are via HTTP, so everything including your RPC login and password are flying around as plaintext. You can use HTTPS and a [self-signed certificate](https://docs.specter.solutions/desktop/self-signed-certificates/) to fix that.
```

### *What do I need to do in order to create a multisig wallet?*

> XPUBs are needed (from HWW's, laptop with Electrum desktop wallet, Specter-DIY, etc.) in order to create a multisig setup, but don't worry it's in watch-only mode! First you need to “add devices” that store keys for the wallet. After creating the devices, you have to create the type of wallet you want (2-of-2, 3-of-5, etc.) and select the corresponding devices/keys - you need at least two devices setup in order to create a multisig wallet.

Reviewer: ```suggestion
XPUBs are needed (from HWW's, laptop with Electrum desktop wallet, Specter-DIY, etc.) in order to create a multisig setup, but don't worry it's in watch-only mode! First you need to “add devices” that store keys for the wallet. After creating the devices, you have to create the type of wallet you want (2-of-2, 3-of-5, etc.) and select the corresponding devices/keys - you need at least two devices to create a multisig wallet.
```

Absolutely, as well as any other DIY bitcoin full, or pruned, node!

> Currently Raspiblitz (https://github.com/rootzoll/raspiblitz), has explicit support and you can automatically install it as bonus-software. Also [umbrel](https://getumbrel.com/) has it in the app-store. Mynode has it on Mynode [premium](https://mynodebtc.com/products/premium). Start9 has it [as well](https://start9labs.medium.com/new-service-specter-ba89f50f27c5). There are differences mainly on update-policy and update-frequency.

Reviewer: ```suggestion
Currently Raspiblitz (https://github.com/rootzoll/raspiblitz), has explicit support and you can automatically install it as bonus-software. Also [umbrel](https://getumbrel.com/) has it in the app store. Mynode has it on Mynode [premium](https://mynodebtc.com/products/premium). Start9 has it [as well](https://start9labs.medium.com/new-service-specter-ba89f50f27c5). There are differences mainly on update policy and frequency.
```

### *Can I use Tor?*

Yes there is Tor support. People are using Tor in two ways: 

> * Increase privacy by routing external calls through Tor

Reviewer: ```suggestion
* To increase privacy by routing external calls through Tor
```

When using Specter and importing an old wallet you would need to re-scan the blockchain in the wallet settings page.

The labeld which you set within Specter will only reappear if you restore a Specter-backup, though.

### *To recover a multisig that was built on specter (eg: 2 of 3 with ColdCard), is having the seeds of all 3 signing wallets sufficient or do we need to backup more info?*

> Having seeds is enough, but in case you lose one of the seeds it is also **mandatory** that you also backup your XPUBs. You can go to the wallet settings and export it as json file, this file has all the information needed to find your funds. "Export to wallet" software should give you one json file with all information needed for the recovery of your watch only wallet later on. 

Reviewer: ```suggestion
Having seeds is enough, but in case you lose one of the seeds it is also **mandatory** that you also backup your XPUBs. You can go to the wallet settings and export it as json file, this file has all the **public** key information needed. "Export to wallet" software should give you one json file with all information needed for the recovery of your watch only wallet later on. 
```

### *Is specter-DIY safe to use?*

> It has been used by many people in production in multi- and singlesig setups. Use it on your own risk and there is no warranty you can break. Don't trust, verify!

Reviewer: ```suggestion
It has been used by many people in production in multi- and singlesig setups. But, keep in mind: "Don't trust, verify!"
```

### *Do you have a physical security design?*

> No security at the moment, but it also doesn't store the private key. Working on integration of secure element similar to the ColdCard's (mikroe secure chip).

Reviewer: ```suggestion
There is physical security, if you use the smart cards, which is a similar setup to Coldcard's usage of secure elements (to store secrets).
```

### *How to delete a wallet using a remote full node?*

You can't delete the wallet if you are using remote Bitcoin Core node - there is no RPC call to do it remotely. So, deleting wallet works only on the same computer. 

> You can also just delete the wallet manually. It's a folder in `~/.bitcoin` directory and in `~/.specter` as well. If you use Spectrum/Electrum as a backend, there is currently no way to delete it. But we're working on that.

Reviewer: ```suggestion
You can just delete the wallet manually. It's in a folder in the Bitcoin directory. Deleting the wallet in the Specter UI always deletes it from your Specter folder. If you use Spectrum as a backend, there is currently nothing in the UI to delete the wallet in the local database. But, we're working on that.
```

## VIDEOS

> Unfortunately, we currently only have videos for Version 1. Version 2 videos are missing. Ping us if you've found any.

Reviewer: ```suggestion
Unfortunately, we currently only have videos for version 1.0. Version 2.0 videos are missing. Ping us, if you find any.
```
